### ISSUE TITLE
Syntax error false positive on nested alternative patterns

### ISSUE THREAD

**Issue by ntBre** [astral-sh/ruff#21101]
This change seems to now cause a SyntaxError on the following valid code that doesn't use different names:

```
invalid-syntax: alternative patterns bind different names
   --> src/sp_repo_review/checks/ruff.py:127:17
    |
125 |           match ruff:
126 |               case (
127 | /                 {"lint": {"select": x} | {"extend-select": x}}
128 | |                 | {"select": x}
129 | |                 | {"extend-select": x}
    | |______________________________________^
130 |               ):
131 |                   return cls.code in x or "ALL" in x
    |
```

See https://github.com/scientific-python/cookie/pull/663

_Originally posted by @henryiii in https://github.com/astral-sh/ruff/issues/20682#issuecomment-3452968119_

and here's a minimized [example](https://play.ruff.rs/f0eab842-e43c-4142-a764-df78b8c078ed):

```py
match ruff:
    case {"lint": {"select": x} | {"extend-select": x}} | {"select": x}:
        return cls.code in x or "ALL" in x
```
